lib/action_account.rb

```diff
@@ -0,0 +1,239 @@
+require 'optparse'
+
+class ActionAccount
+  attr_reader :argv, :stdout, :stderr
+
+  def initialize(argv:, stdout:, stderr:)
+    @argv   = argv
+    @stdout = stdout
+    @stderr = stderr
+  end
+
+  Result = Struct.new(
+    :table,   # tabular output, rendered as an ASCII table or as CSV
+    :json,    # output that should only be formatted as JSON
+    :subtask, # name of subtask, used for audit logging
+    :uuids,   # Array of UUIDs entered or returned, used for audit logging
+  )
+
+  Config = Struct.new(
+    :include_missing,
+    :format,
+    :show_help,
+  ) do
+    alias_method :include_missing?, :include_missing
+    alias_method :show_help?, :show_help
+  end
+
+  def config
+    @config ||= Config.new(
+      include_missing: true,
+      format: :table,
+      show_help: false,
+    )
+  end
+
+  def run
+    option_parser.parse!(argv)
+    subtask_class = subtask(argv.shift)
+
+    if config.show_help? || !subtask_class
+      stderr.puts '*Task*: `help`'
+      stderr.puts '*UUIDs*: N/A'
+
+      stdout.puts option_parser
+      return
+    end
+
+    result = subtask_class.new.run(args: argv, config:)
+
+    stderr.puts "*Task*: `#{result.subtask}`"
+    stderr.puts "*UUIDs*: #{result.uuids.map { |uuid| "`#{uuid}`" }.join(', ')}"
+
+    if result.json
+      stdout.puts result.json.to_json
+    else
+      render_output(result.table)
+    end
+  end
+
+  # @param [Array<Array<String>>] rows
+  def render_output(rows)
```

Comment:
I'm a little torn, this is identical to the output formatting in the Data Pull script, do you think we should pull this shared code into a helper class and delegate out it?